### ISSUE TITLE
Plugin update broke site

### ISSUE THREAD

**Issue by ghost** [woocommerce/woocommerce-paypal-payments#2654]
Did one at a time updates of Woocommerce, Woopayments, and finally WooCommerce PayPal Payments.  WooCommerce PayPal Payments broke the website- can't access the site or wordpress admin area and have to restore from backups
Error Text: "There has been a critical error on this website. Please check your site admin email inbox for instructions.

[Learn more about troubleshooting WordPress.](https://wordpress.org/documentation/article/faq-troubleshooting/)"

### To Reproduce

1. Go to '…'
2. Click on '…'
3. Scroll down to …'

### Environment

 - WordPress Version: Current version: 6.6.2
 - WooCommerce Version: (after backup restore - first I updated all plugins and got the critical failure. then i updated woocommerce and woopayments successfully first before updating woopayments paypal and getting the critical failure)
 - Plugin Version: 9.3.2
 - Browser [e.g. Chrome, Safari] and Version:Firefox version 130.0.1 (64-bit) 
 - Any other plugins installed:

"<html><body>

You have version 3.24.3 installed. Update to 3.24.4. View version 3.24.4 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
--
Select Essential Blocks | Essential Blocks 			You have version 5.0.0 installed. Update to 5.0.1. View version 5.0.1 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select Image Optimizer by Elementor – Compress, Resize and Optimize Images | Image Optimizer by Elementor – Compress, Resize and Optimize Images 			You have version 1.5.2 installed. Update to 1.5.3. View version 1.5.3 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select Really Simple SSL | Really Simple SSL 			You have version 9.0.0 installed. Update to 9.0.2. View version 9.0.2 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select ShopLentor – WooCommerce Builder for Elementor & Gutenberg | ShopLentor – WooCommerce Builder for Elementor & Gutenberg 			You have version 2.9.7 installed. Update to 2.9.8. View version 2.9.8 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select UpdraftPlus - Backup/Restore | UpdraftPlus - Backup/Restore 			You have version 1.24.4 installed. Update to 1.24.6. View version 1.24.6 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select WooCommerce | WooCommerce 			You have version 9.3.2 installed. Update to 9.3.3. View version 9.3.3 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select WooCommerce PayPal Payments | WooCommerce PayPal Payments 			You have version 2.9.0 installed. Update to 2.9.1. View version 2.9.1 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select WooPayments | WooPayments 			You have version 8.2.1 installed. Update to 8.2.2. View version 8.2.2 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select WPForms Lite | WPForms Lite 			You have version 1.9.0.4 installed. Update to 1.9.1.2. View version 1.9.1.2 details.Compatibility with WordPress 6.6.2: 100% (according to its author)
Select YITH WooCommerce Subscription | YITH WooCommerce Subscription 			You have version 4.1.0 installed. Update to 4.1.1. View version 4.1.1 details.

</body>
</html>You have version 3.24.3 installed. Update to 3.24.4. [View version 3.24.4 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=elementor&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select Essential Blocks
	

Essential Blocks
You have version 5.0.0 installed. Update to 5.0.1. [View version 5.0.1 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=essential-blocks&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select Image Optimizer by Elementor – Compress, Resize and Optimize Images
	

Image Optimizer by Elementor – Compress, Resize and Optimize Images
You have version 1.5.2 installed. Update to 1.5.3. [View version 1.5.3 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=image-optimization&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select Really Simple SSL
	

Really Simple SSL
You have version 9.0.0 installed. Update to 9.0.2. [View version 9.0.2 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=really-simple-ssl&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select ShopLentor – WooCommerce Builder for Elementor & Gutenberg
	

ShopLentor – WooCommerce Builder for Elementor & Gutenberg
You have version 2.9.7 installed. Update to 2.9.8. [View version 2.9.8 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=woolentor-addons&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select UpdraftPlus - Backup/Restore
	

UpdraftPlus - Backup/Restore
You have version 1.24.4 installed. Update to 1.24.6. [View version 1.24.6 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=updraftplus&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select WooCommerce
	

WooCommerce
You have version 9.3.2 installed. Update to 9.3.3. [View version 9.3.3 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=woocommerce&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select WooCommerce PayPal Payments
	

WooCommerce PayPal Payments
You have version 2.9.0 installed. Update to 2.9.1. [View version 2.9.1 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=woocommerce-paypal-payments&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select WooPayments
	

WooPayments
You have version 8.2.1 installed. Update to 8.2.2. [View version 8.2.2 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=woocommerce-payments&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select WPForms Lite
	

WPForms Lite
You have version 1.9.0.4 installed. Update to 1.9.1.2. [View version 1.9.1.2 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=wpforms-lite&section=changelog&TB_iframe=true&width=640&height=662)
Compatibility with WordPress 6.6.2: 100% (according to its author)
Select YITH WooCommerce Subscription
	

YITH WooCommerce Subscription
You have version 4.1.0 installed. Update to 4.1.1. [View version 4.1.1 details.](https://vieques.coffee/wp-admin/plugin-install.php?tab=plugin-information&plugin=yith-woocommerce-subscription&section=changelog&TB_iframe=true&width=640&height=662)"

### Additional Details

<details><summary>System status</summary>

  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```

</details>

**Comment by bwebnaomi** [comment]
Any luck on this? I've got the same problem.

**Comment by bwebnaomi** [comment]
I found this article inspiring - https://wordpress.org/support/topic/woocommerce-paypal-payments-version-2-9-1-fatal-error/

I was using a snippet to change the pay later functionality into the legacy mode. After removing that code it all worked again nicely. 

To get it all working again 

1. I changed the name of the Paypal plugin (temporarily) via FTP
2. removed the snippet, 
3. changed the name back to the original name and 
4. then activated the plugin again

Hope this helps

**Comment by InpsydeNiklas** [comment]
Hi @nighthawkmountain, please update the `YITH WooCommerce Subscription` plugin to the latest version 4.1.1 as this added support for the latest PayPal Payments version: https://wordpress.org/plugins/yith-woocommerce-subscription/#developers
We apologize for the error after updating PayPal Payments. If any issues remain after this update, please [contact our support team from here](https://woocommerce.com/document/woocommerce-paypal-payments/#get-help). Thank you!

@bwebnaomi the issue you faced is a different one, but it was addressed in the upcoming 2.9.2 update: https://github.com/woocommerce/woocommerce-paypal-payments/releases/tag/2.9.2-rc1
However, we do recommend keeping the new Pay Later messaging configurator enabled (not using the code snippet) as it should work and save your settings as expected by now. However, as mentioned above, please get in touch with our support team if any doubts remain. Again, apologies for the error! Thank you!
